internal/bucket/versioning/versioning.go

```diff
@@ -62,11 +86,47 @@ func (v Versioning) Enabled() bool {
 	return v.Status == Enabled
 }
 
+// PrefixEnabled - returns true if versioning is enabled at the bucket and given
+// prefix, false otherwise.
+func (v Versioning) PrefixEnabled(prefix string) bool {
+	if v.Status == Enabled {
```

Comment:
```suggestion
	if v.Status != Enabled {
	       return false
        }
```

How about we invert the logic here to avoid a big enough `if else` tabbing - this is a minor nit, feel free to ignore.